openmetadata-service/src/main/resources/elasticsearch/en/table_index_mapping.json

```diff
@@ -99,6 +99,7 @@
         "normalizer": "lowercase_normalizer"
       },
       "columns": {
+        "type": "nested",
```

Comment:
can we check the column name aggregate

Reply:
Hey @harshach , great catch. With this change it actually breaks the column search.

I was searching for alternatives and if we also add the `include_in_parent = true` attribute it seems to fix the search. https://www.elastic.co/guide/en/elasticsearch/reference/current/nested.html

I'm a bit worried that basically this means we're saving this information twice. Once on its own index and the other one as it currently is.

Was also thinking about other alternatives that we could use instead of changing the mapping type:

- [Script Query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-script-query.html): It's said to have a bad performance and I couldn't access the fields that I actually needed to access since they are analyzed text fields. We could technically change that but it [doesn't seem to be a good way](https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting-fields.html#modules-scripting-doc-vals)

- Having extra meta-metadata fields that we could calculate internally and use it for filtering.

What do you think?

Thanks

Reply:
Should we add a test around this use case @IceS2?

Reply:
While talking with @ShaileshParmar11 on this task, he mentioned that we've got a really nice aggregate we use to fetch the filter list and that it also shows the amount of documents with that specific filter.

With the proposed solution I saw that it weasn't really possible to do that since we're doing some hacky wildcard search on text fields and I was trying to change it.

I learnt a bit more about ElasticSearch and thought about adding some runtime fields and we could have something like this:

```    
"runtime":  {
      "descriptionStatus": {
        "type": "keyword",
        "script": {
          "source": "int columnQty = 0; int columnDescriptionQty = 0; if (doc.containsKey('columnNames')) { columnQty = doc['columnNames'].length; } if (params['_source'].containsKey('columns')) { for (column in params['_source']['columns']) { if (column.containsKey('description') && column['description'] != '') { columnDescriptionQty += 1; } }} if (columnDescriptionQty == columnQty && params['_source'].containsKey('description') &&params['_source']['description'] != '') {  emit('COMPLETE')} else {  emit('INCOMPLETE')}"
        }
      },
      "ownerKeywordWithNoOwner": {
        "type": "keyword",
        "script": {
          "source": "if (doc.containsKey('owner.name.keyword') && doc['owner.name.keyword'].size() > 0) { emit(doc['owner.name.keyword'].value) } else { emit('No Owner') }"
        }
      }
},
```

I believe we could define the fields on the mapping itself for each asset instead of relying on a runtime field at query time since each asset might have different conditions.

What do you think?

Reply:
@IceS2 any reason to have these as runtime fields , why not pre-compute them in the index docs TableIndex.java etc..

Reply:
Only reason is me not being familiar with the code yet ;P

That's probably the best solution (And fyi, Opensearch does not support runtime mappings at the time being)